cowprotocol/services
# Maybe a bit of a simplification for the incremental cache

# Description
My attempt to hopefully simplify one function of https://github.com/cowprotocol/services/pull/2923 slightly.

# Changes
Mostly tried to condense the code a bit and make timestamp handling easier to understand.
I believe the only logical change was that I initialize the timestamp with a non-zero value on the first iteration. However, IIUC since we now use the incremental cache every time when the cache is `Some` this should cancel each other out making the whole thing a bit easier to grasp IMO.

## How to test
e2e tests still pass (tests ran fine locally but 1 test fails in CI, will investigate to see if it's the PR or a flaky test in general :/)
@squadgazzz could you give this a sanity check that it's actually not breaking any logic?
